synapse/push/pusherpool.py

```diff
@@ -281,14 +286,14 @@ async def start_pusher_by_id(
 
         resultlist = await self.store.get_pushers_by_app_id_and_pushkey(app_id, pushkey)
 
-        pusher_dict = None
+        pusher_config = None
         for r in resultlist:
-            if r["user_name"] == user_id:
-                pusher_dict = r
+            if r.user_name == user_id:
+                pusher_config = r
 
         pusher = None
-        if pusher_dict:
-            pusher = await self._start_pusher(pusher_dict)
+        if pusher_config:
+            pusher = await self._start_pusher(pusher_config)
```

Comment:
This logic seems inefficient since there's a `get_pushers_by` which could also accept the `user_name` parameter. It seemed unrelated to this PR though so I haven't pushed a change to use that.